### ISSUE TITLE
First frame is not updating after switching workspaces

### ISSUE THREAD

**Issue by oifj34f34f** [swaywm/sway#7696]
- **Sway Version:**
  - 1.8.1, also tested on 1.9-dev-df8aad9c

- **Debug Log:**
  - https://pastebin.com/raw/TC41Uuyn

- **Demonstration:**
  - https://0x0.st/H2eu.mp4 (please use mpv)

- **To Reproduce:**
  - Open any program.
  - Switch to another workspace.
  - Make changes in the program window by e.g. switching the track if it's an audio player.
  - Switch back to first workspace and see the first frame (bug).

- **Description:**
  - As you can see from the demo, after switching to the first workspace, the first frame of the program window shows "3" instead of "7" (slow down the video).

**Comment by lukoramu** [comment]
Yes, on AMD GPU, sway version 1.11 this 'single old frame showing' phenomenon is present.

I also see it when switching tabs inside the same workspace, not only when switching workspaces.

I imagine there should be an intermediate state where compositor initiates frame callbacks to all the required applications we're about to display, and then complete the switch on the next scanout... I think... :-)
